src/System.Net.Http/src/System/Net/Http/HttpContent.cs

```diff
@@ -629,6 +635,11 @@ private static bool BufferHasPrefix(ArraySegment<byte> buffer, byte[] prefix)
             return returnFunc(state);
         }
 
+        private static Exception CreateOverCapacityException(int maxBufferSize)
+        {
+            throw new HttpRequestException(SR.Format(SR.net_http_content_buffersize_exceeded, maxBufferSize));
```

Comment:
Ugh, yes :)  Either that or change the return type and don't have the call sites throw.  Will fix.